### ISSUE TITLE
Remove aria-labelledby when there's no label

### ISSUE THREAD

**Issue by anezthes** [vaadin/vaadin-text-field#562]
The `aria-labelledby` attribute should be removed when there's no label.

In addition, VoiceOver does not announce the `aria-label` on Chrome unless the `aria-labelledby` attribute is removed.

**Current workaround (Flow)**
```
TextField field = new TextField();
field.getElement().executeJs("this.inputElement.setAttribute('aria-label', 'First name')");
field.getElement().executeJs("this.inputElement.removeAttribute('aria-labelledby')");
```

**Desired workflow**
```
TextField field = new TextField();
field.setAriaLabel("First name");
```

**Related issues**
* https://github.com/vaadin/vaadin-text-field/issues/454
* https://github.com/vaadin/vaadin-text-field/issues/473

**Comment by web-padawan** [comment]
Please see https://github.com/vaadin/vaadin-text-field/issues/562

> When the focused part is not an input field, e. g. in `<vaadin-select>`, setting `aria-labelledby` with only the label id alone makes the value not announced. Including both label and value container ids helps.

This is the description from the related commit a11bc6a85fc056533dfecdaed926c7687998ccff by @platosha 

IIRC it was done for `vaadin-select` so I guess we could move the related logic there.

**Comment by rolfsmeds** [comment]
We already have a ticket for adding API for aria-label: https://github.com/vaadin/vaadin-text-field/issues/454
